### ISSUE TITLE
Possible loss of precision for GpsTime stats

### ISSUE THREAD

**Issue by wonder-sk** [hobu/untwine#85]
When using untwine to index a file with a small range of GpsTime values, the minimum and maximum values loose precision when written to ept.json:
```json
{
  "name": "GpsTime",
  "type": "float",
  "size": 8,
  "count": 53561,
  "maximum": 2.29284e+08,
  "minimum": 2.29284e+08,
  "mean": 2.29284e+08,
  "stddev": 0.519126,
  "variance": 0.269492
}
```
It would be nice if longer precision would be enforced, because based on that QGIS thinks the min/max are the same:

![image](https://user-images.githubusercontent.com/193367/136230525-fa8c15e6-2798-4f83-a46b-9a78bb65bf84.png)

Test file: [test-file-gpstime.zip](https://github.com/hobu/untwine/files/7295099/test-file-gpstime.zip)

**Comment by abellgithub** [comment]
This just isn't a thing for double precision values, which these are. There
is no magic way to figure out the right number of digits, which is why
C/C++ has all these options. But I really don't want to argue about this
and I'll just take care of the issue at hand.

**Comment by abellgithub** [comment]
This is what I mean:

#include <iomanip>
#include <iostream>
#include <limits>

int main()
{
    double d = .1;

    std::cout << std::setprecision(17) << d << "\n";
    std::printf("%.*g\n", std::numeric_limits<double>::max_digits10, d);
    return 0;
}

$ a.out
0.10000000000000001
0.10000000000000001

**Comment by wonder-sk** [comment]
To your questions/notes:
- I didn't mean that GpsTime should be treated specially, just reported it as a concrete problem I have come across
- lossless representation of all numbers in ept.json would be the best solution in my opinion
- using precision of `std::numeric_limits<double>::max_digits10` seems like the correct approach if minimal changes should be done. Personally I don't see a problem if `0.1` gets written as `0.10000000000000001`, as long as the reader will parse it as the same number
- nlohmann/json is worth trying - we have stared to use it in QGIS as well instead of Qt's JSON classes

**Comment by abellgithub** [comment]
People have whined when they see something like 0.1000000000001. GDAL has a ton of awful code to deal with such things because people have whined. I don't care about such things, but there is history of bitching. The point was that there is no "right". It depends on what you want and you can't easily tell what someone wants, because it usually depends on what they see when they actually look at a file. When people type in ".1" for a scale factor, they expect to see exactly that at output.  But at other times they write something like "123456.789012345" and when a digit gets dropped, they also whine. There is no winning.

**Comment by abellgithub** [comment]
Agree. That's what I'll do.

I think a lot of the complaints in GDAL were WRT WKT geometry and geojson where people were concerned about file size.

**Comment by abellgithub** [comment]
Fixed by #105
